### PR TITLE
Make test code use TestEnvironment

### PR DESCRIPTION
**Pull request by veblush**
Updated unit tests to call `grpc_shutdown_blocking` at the end of test so that there is no active thread at the time when the main function exits. Without this, memory sanitizer won't be happy.

`grpc_shutdown_blocking` can be called in two ways.
- Explicitly calling `grpc_shutdown_blocking` instead of `grpc_shutdown`
- Using `grpc::testing::TestEnvironment` which waits for the shutdown thread to finish at its destructor. 

Related to #21606

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10125,6 +10125,258 @@ if(gRPC_BUILD_TESTS)
 
 add_executable(alts_grpc_record_protocol_test
   test/core/tsi/alts/zero_copy_frame_protector/alts_grpc_record_protocol_test.cc
+  src/core/ext/filters/client_channel/resolver/fake/fake_resolver.cc
```
Comment (yang-g) [inline CMakeLists.txt]: Is this change intentional?

Comment (veblush) [reply CMakeLists.txt]: Yes, this change showed up because it needs to have new dependency against `grpc_test_util_base` to use `grpc::testing::TestEnvironment`
